packages/langium/src/grammar/validation/validator.ts

```diff
@@ -411,6 +413,45 @@ export class LangiumGrammarValidator {
         }
     }
 
+    checkEmptyParserRule(parserRule: ast.ParserRule, accept: ValidationAcceptor): void {
+        // Rule body needs to be set.
+        // Entry rules and fragments may consume no input.
+        if (!parserRule.definition || parserRule.entry || parserRule.fragment) {
+            return;
+        }
+
+        // Process definition; report problem for the rule itself.
+        const mayConsumeEmpty = (element: ast.AbstractElement): boolean => {
+            // First, check cardinality of the element.
+            if (element.cardinality === '?' || element.cardinality === '*') {
+                return true;
+            }
+            // Actions themselves count as non-optional.
+            if (ast.isAction(element)) {
+                return false;
+            }
+            // Assignments with ?= consume nothing, groups without + too.
+            if (ast.isAssignment(element)) {
+                return element.operator === '?=';
+            }
+            // Unordered groups consume something.
+            if (ast.isUnorderedGroup(element)) {
+                return false;
+            }
+            // Ordered groups consume if all elements do so;
+            // and alternative groups, too.
+            if (ast.isGroup(element) || ast.isAlternatives(element)) {
+                return element.elements.every(mayConsumeEmpty);
+            }
+            // Else, assert that we consume _something_.
+            return false;
+
+        };
+        if (mayConsumeEmpty(parserRule.definition)) {
+            accept('error', 'This parser rule would succeed without consuming input.', { node: parserRule, property: 'definition', code: IssueCodes.ParsingRuleEmpty });
```

Comment:
Suggestion: I would use the `name` property to display the message. Diagnostics should rarely cover the whole definition of a node, but rather a very specific part of it. For example, if I forgot to implement a method of an interface, I don't get an error on the whole class body, but on its name instead.

Reply:
Now using the 'name' property.